lib/puppet/ssl/host.rb

```diff
@@ -568,14 +568,14 @@ def check_for_certificate_on_disk(cert_name)
   def download_certificate_from_ca(cert_name)
     return nil if Puppet::SSL::Host.ca_location != :remote
 
-    response = Puppet::Rest::Routes.get_certificate(http_client, cert_name)
-    if response.ok?
-      body = response.read_body
-      begin
-        Puppet::SSL::Certificate.from_s(body)
-      rescue OpenSSL::X509::CertificateError
-        raise Puppet::Error, _("Response from the CA did not contain a valid certificate for %{cert_name}.") % { cert_name: cert_name }
-      end
+    begin
```

Comment:
In `download_ca_certificate_bundle` we nest begin/resuce, where here we don't. We should probably do the same in each place? I would guess nesting in more correct?

Reply:
Updated with nesting.